test/e2e/new-dns-record.spec.ts

```diff
@@ -0,0 +1,40 @@
+import { test, expect } from '@playwright/test';
+
+test.describe('not authenticated', () => {
+  test('redirects to login page', async ({ page }) => {
+    await page.goto('/');
+    await expect(page).toHaveURL(/.*login.*/);
+  });
+});
+
+test.describe('authenticated as user', () => {
+  // Use shared state for being logged in as user
+  test.use({ storageState: 'test/e2e/.auth/user.json' });
+
+  test('redirects to edit dns record page when required fields are filled', async ({ page }) => {
+    await page.goto('/domains/new');
+    await page.getByLabel('Record Name*').fill('test');
+    await page.getByRole('combobox', { name: 'Type' }).selectOption('A');
+    await page.getByLabel('Value*').fill('test');
+    await page.getByRole('button', { name: 'Create' }).click();
+    await expect(page).toHaveURL(/.*domains\/[0-9]*/);
+  });
+
+  test('redirects to edit dns record page when all fields are filled', async ({ page }) => {
+    await page.goto('/domains/new');
+    await page.getByLabel('Record Name*').fill('test');
+    await page.getByRole('combobox', { name: 'Type' }).selectOption('A');
+    await page.getByLabel('Value*').fill('test');
+    await page.getByLabel('Ports').fill('port1, port2');
+    await page.getByLabel('Course').fill('test course');
+    await page.getByLabel('Description').fill('test description');
+    await page.getByRole('button', { name: 'Create' }).click();
+    await expect(page).toHaveURL(/.*domains\/[0-9]*/);
```

Comment:
Question: do we not need to wait on a navigation here?  Or is the fake routing we're doing in Remix such that we'll not need to worry about it?

Reply:
We don't need to worry about it because the `toHaveURL` should wait until the page navigates to that URL, up to the configured timeout seconds